src/model.cpp

```diff
@@ -373,10 +393,16 @@ Model::load_3ds(const std::string &filename)
     while (!input_file.eof()) {
         uint16_t chunk_id;
         uint32_t chunk_length;
-
+		uint32_t swap32t;
+		uint16_t swap16t;
+		
         // Read the chunk header
         input_file.read(reinterpret_cast<char *>(&chunk_id), 2);
-        if (input_file.gcount() == 0) {
+		if(is_bigendian()) {
+			swap16t=__bswap_16(chunk_id); chunk_id=swap16t;
```

Comment:
The byteswap.h API is `bswap_16/32/64()`. The underscored versions are internal implementation details.

Also, AFAIK, there is no need for an intermediate variable. We could more simply write: `chunk_id = is_bigendian() ? bswap_16(chunk_id) : chunk_id;` (like it's already done for some cases below).

Similarly for all swaps below.